test/flattenProp.js

```diff
@@ -0,0 +1,65 @@
+import RA from '../src/index';
+import eq from './shared/eq';
+
+
+describe('flattenProp', function() {
+  let prop;
+  let obj;
+
+  beforeEach(function () {
+    prop = 'b';
+    obj = {
+      a: 1,
+      b: { c: 3, d: 4 },
+    };
+  });
+
+  it('tests currying', function () {
+    eq(RA.flattenProp('prop', {}), {});
+    eq(RA.flattenProp('prop')({}), {});
+  });
+
+  context("when prop doesn't exist", function () {
+    specify('should return object with identical structure as provided object', function () {
+      eq(RA.flattenProp('not_exist', obj), obj);
+    });
+  });
+
+  context('when prop leads to non object', function () {
+    specify('should return object with identical structure as provided object', function () {
+      obj = {
+        a: 1,
+        b: 999,
+      };
+      eq(RA.flattenProp(prop, obj), obj);
+    });
+  });
+
+  context('when prop leads to object with same prop name', function () {
+    specify('should override existing props on provided object', function () {
+      obj = {
+        a: 1,
+        b: { b: 999, c: 3, d: 4 },
+      };
+      const expected = {
+        a: 1,
+        c: 3,
+        d: 4,
+        b: 999,
+      };
+      eq(RA.flattenProp(prop, obj), expected);
+    });
+  });
+
+  context('when standard usecase', function () {
```

Comment:
We can drop context here IMHO and rename specify to simple `it`. The fact that it is standard usecase is obvious from the test it self.